### DISCUSSION TITLE
Soapheader?

### DISCUSSION THREAD

**Discussion by linusalnervik** [CoreWCF/CoreWCF#650] [Q&A] [answered]
Hi. 
I am trying to get the custom values inserted in to the soapheader in the request.

`[Injected] HttpRequest httpRequest`
gives me the http headers with
` httpRequest.Headers`
but they do not contain the cusom soapheader.

What am i missing here?

**Comment by g7ed6e** [comment, marked as answer]
`HttpRequest` is an ASP.NET Core abstraction thus it does not know SOAP headers. 
Did you try using `MessageContract` instead of `DataContract` ? I think it is supported in CoreWCF too, please see https://docs.microsoft.com/en-us/dotnet/framework/wcf/feature-details/using-message-contracts
